File: src/main/java/corp/soyummy/recipes/dto/RecipeCreateDto.java
package corp.soyummy.recipes.dto;

import lombok.Getter;
import lombok.Setter;
import lombok.AllArgsConstructor;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

import java.util.List;
import corp.soyummy.recipes.RecipeIngredient;

@Getter
@Setter
@AllArgsConstructor
public class RecipeCreateDto {
    @NotNull(message = "Title is required!")
    @NotBlank(message = "Title is required!")
    private String title;

    @NotNull(message = "Category is required!")
    @NotBlank(message = "Category is required!")
    private String category;

    @NotNull(message = "Area is required!")
    @NotBlank(message = "Area is required!")
    private String area;

    @NotNull(message = "Instructions is required!")
    @NotBlank(message = "Instructions are required!")
    private String instructions;

    @NotNull(message = "Description is required!")
    @NotBlank(message = "Description is required!")
    private String description;

    @Size(min = 1, message = "Time must have at least one character.")
    private String time;

    @Pattern(regexp = "^https://.{3,}", message = "The URL must start with \"https://\" and have at least 3 characters following.")
    private String thumbnail;

    @Pattern(regexp = "^https://.{3,}", message = "The URL must start with \"https://\" and have at least 3 characters following.")
    private String preview;

    @Pattern(regexp = "^https://.{3,}", message = "The URL must start with \"https://\" and have at least 3 characters following.")
    private String youtube;

    private List<String> tags;
    private List<RecipeIngredient> ingredients;
}

File: src/main/kotlin/com/soyummy/recipes/dto/RecipeCreateDto.kt
package com.soyummy.recipes.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import com.soyummy.recipes.RecipeIngredient

data class RecipeCreateDto(
    @field:NotBlank(message = "Title is required!")
    val title: String,

    @field:NotBlank(message = "Category is required!")
    val category: String,

    @field:NotBlank(message = "Area is required!")
    val area: String,

    @field:NotBlank(message = "Instructions are required!")
    val instructions: String,

    @field:NotBlank(message = "Description is required!")
    val description: String,

    @field:NotBlank(message = "Time is required!")
    val time: String,

    @field:Pattern(regexp = "^https://.{3,}", message = "The URL must start with \"https://\" and have at least 3 characters following.")
    val thumbnail: String,

    @field:Pattern(regexp = "^https://.{3,}", message = "The URL must start with \"https://\" and have at least 3 characters following.")
    val preview: String,

    @field:Pattern(regexp = "^https://.{3,}", message = "The URL must start with \"https://\" and have at least 3 characters following.")
    val youtube: String,

    val tags: List<String>? = null,
    val ingredients: List<RecipeIngredient>? = null
)